scanner/src/main/kotlin/provenance/NestedProvenanceScanResult.kt

```diff
@@ -130,6 +133,27 @@ data class NestedProvenanceScanResult(
         return findings
     }
 
+    private fun Map<KnownProvenance, List<ScanResult>>.mergeSnippetFindings(): Set<SnippetFinding> {
+        val findingsByPath = mapKeys { getPath(it.key) }.mapValues { (_, scanResults) ->
+            mutableSetOf<SnippetFinding>().also { acc ->
+                scanResults.forEach { acc += it.summary.snippetFindings }
+            }
+        }
+
+        val allFindings = mutableSetOf<SnippetFinding>()
+
+        findingsByPath.map { (path, findings) ->
+            val prefix = if (path.isEmpty()) path else "$path/"
+
+            allFindings += findings.map { finding ->
+                val newPath = "$prefix${finding.sourceLocation.path}"
+                finding.copy(sourceLocation = finding.sourceLocation.copy(path = newPath))
+            }.toSet()
```

Comment:
Remove `toSet()` because there is no need to convert to set before adding the collection to a set.